webapps/frontend/app/viewmodels/services/storagerouter.js

```diff
@@ -47,7 +46,34 @@ define([
             return api.post('storagerouters/' + storageRouterGuid + '/get_metadata')
                 .then(shared.tasks.wait)
         };
-
+        /**
+         * Fetches support info of a StorageRouter
+         * @param storageRouterGuid: Guid of the StorageRouter
+         * @return {Promise<T>}
+         */
+        self.getSupportSettingsHandle = function(storageRouterGuid) {
+            return api.get('storagerouters/' + storageRouterGuid + '/get_support_info')
+                .then(shared.tasks.wait)
+        };
+        /**
+         * Fetches support metadata of a StorageRouter
+         * @param storageRouterGuid: Guid of the StorageRouter
+         * @return {Promise<T>}
+         */
+        self.getSupportMetadataHandle = function(storageRouterGuid) {
+            return api.get('storagerouters/' + storageRouterGuid + '/get_support_metadata')
+                .then(shared.tasks.wait)
+        };
+        /**
+         * Fetches support info of a StorageRouter
+         * @param storageRouterGuid: Guid of the StorageRouter
+         * @param metaData: data to post
+         * @return {Promise<T>}
+         */
+        self.postSupportData = function(storageRouterGuid, metaData) {
```

Comment:
I'd call it saveSupportdata.
Also it is not metadata but data to. Rename the parameter to reflect that.